src/puppetlabs/pcp/client.clj

```diff
@@ -111,7 +111,9 @@
     (let [sessions (.getOpenSessions websocket-client)]
       (log/debug (i18n/trs "Sending WebSocket ping"))
       (doseq [session sessions]
-        (.. session (getRemote) (sendPing (ByteBuffer/allocate 1))))))
+        ;; sessions in the CLOSING state are in OpenSessions
+        (when (.isOpen session)
+          (.. session (getRemote) (sendPing (ByteBuffer/allocate 1)))))))
```

Comment:
This has a race, where it could start CLOSING after the call to isOpen. Handling the exception might be a better option. I thought we were allowing that to happen and cause the thread to terminate actually.